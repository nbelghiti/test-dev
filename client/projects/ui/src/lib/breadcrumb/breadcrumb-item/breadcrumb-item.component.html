<div uiFlyout *ngIf="items && items.length > 1; else asLink">
	<button
		uiFlyoutToggle
		class="c-breadcrumb-item"
		[ngClass]="{
			'c-breadcrumb-item--first': first
		}"
		[title]="value"
	>
		<span class="c-breadcrumb-item__label">
			{{label}}
		</span>

		<div class="c-breadcrumb-item__item">
			<span
				*ngIf="conflict"
				class="c-breadcrumb-item__icon"
			>
				<ui-icon
					[name]="IconName.exclamationCircle"
					[size]="IconSize.tiny"
					[type]="IconType.error"
				>
				</ui-icon>
			</span>

			<span class="c-breadcrumb-item__value">
				{{value}}
			</span>

			<span class="c-breadcrumb-item__arrow">
				<ui-icon
					[name]="IconName.caretDown"
					[size]="IconSize.small"
				>
				</ui-icon>
			</span>
		</div>
	</button>

	<ui-filter-menu
		uiFlyoutContent
		class="c-flyout__content--left"
		label="value"
		[items]="items"
		[formControl]="activeBreadcrumb"
	>
	</ui-filter-menu>
</div>

<ng-template #asLink>
	<a
		class="c-breadcrumb-item"
		[ngClass]="{
			'c-breadcrumb-item--first': first
		}"
		[routerLink]="link"
		[queryParams]="queryParams"
		[title]="value"
	>
		<span class="c-breadcrumb-item__label">
			{{label}}
		</span>

		<div class="c-breadcrumb-item__item">
			<span
				*ngIf="conflict"
				class="c-breadcrumb-item__icon"
			>
				<ui-icon
					[name]="IconName.exclamationCircle"
					[size]="IconSize.tiny"
					[type]="IconType.error"
				>
				</ui-icon>
			</span>

			<span class="c-breadcrumb-item__value">
				{{value}}
			</span>
		</div>
	</a>
</ng-template>
