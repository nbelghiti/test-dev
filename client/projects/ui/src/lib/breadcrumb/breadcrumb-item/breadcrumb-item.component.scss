// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Breadcrumb item
// =============================================================================
.c-breadcrumb-item {
	position: relative;

	display: inline-flex;
	flex-direction: column;
	padding: 1.4rem 2rem;

	text-decoration: none;

	border-right: $border-width-base solid $border-color-grey-86;
	color: $text-color-grey-46;

	&:hover {
		color: $text-color-base;

		cursor: pointer;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-breadcrumb-item__item {
	display: inline-flex;
	align-items: center;

	margin-top: 0.2rem;
}

.c-breadcrumb-item__label {
	@include extend-text-overflow();

	display: inline-block;
	max-width: 20rem;

	font-size: $font-size-2x-small;
	line-height: $line-height-normalize;
	text-align: left;
	text-transform: uppercase;

	white-space: nowrap;
}

.c-breadcrumb-item__value {
	@include extend-text-overflow();

	display: inline-block;
	max-width: 20rem;

	font-size: $font-size-1x-small;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;

	white-space: nowrap;
}

.c-breadcrumb-item__arrow {
	display: inline-block;
	margin-left: 1rem;
}

.c-breadcrumb-item__icon {
	display: inline-block;
	margin-right: 0.4rem;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-breadcrumb-item--first {
	background-color: $background-color-grey-96;
}
