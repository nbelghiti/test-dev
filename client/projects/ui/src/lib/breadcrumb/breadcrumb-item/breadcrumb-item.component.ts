import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
import { FormControl } from '@angular/forms';

import { IFilterMenuItem } from '../../filter/filter-menu/filter-menu.component.types';
import { IconName, IconSize, IconType } from '../../icon/icon.types';

@Component({
	selector: 'ui-breadcrumb-item',
	templateUrl: './breadcrumb-item.component.html',
	styleUrls: ['./breadcrumb-item.component.scss']
})
export class BreadcrumbItemComponent implements OnChanges {
	@Input() public label: string = '';
	@Input() public value: string = '';
	@Input() public items: IFilterMenuItem[] = null;
	@Input() public first?: boolean = false;
	@Input() public link?: string[];
	@Input() public conflict?: boolean = false;
	@Input() public queryParams?: { [key: string]: string };

	public IconName: typeof IconName = IconName;
	public IconSize: typeof IconSize = IconSize;
	public IconType: typeof IconType = IconType;
	public activeBreadcrumb: FormControl = new FormControl();

	public ngOnChanges(changes: SimpleChanges): void {
		if (changes.value && changes.value.currentValue) {
			this.activeBreadcrumb.setValue(
				{ value: changes.value.currentValue },
				{ emitEvent: false }
			);
		}
	}
}
