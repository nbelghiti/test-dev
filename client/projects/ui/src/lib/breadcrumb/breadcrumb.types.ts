import { IFilterMenuItem } from '../filter/filter-menu/filter-menu.component.types';

export interface IBreadcrumbItem {
	value: string;
	label?: string;
	link?: string[];
	queryParams?: { [key: string]: string };
	items?: IFilterMenuItem[];
}
