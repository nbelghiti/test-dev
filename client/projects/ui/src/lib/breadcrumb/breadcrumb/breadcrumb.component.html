<nav class="c-breadcrumb">
	<div class="c-breadcrumb__content">
		<ul
			class="c-breadcrumb__list"
			[ngClass]="{
				'c-breadcrumb--border': border
		}"
		>
			<li *ngFor="let item of items">
				<ui-breadcrumb-item
					[first]="first"
					[label]="item?.label"
					[value]="item?.value"
					[link]="item?.link"
					[items]="item?.items"
					[queryParams]="item?.queryParams"
				>
				</ui-breadcrumb-item>
			</li>
		</ul>

		<ui-button
			*ngIf="editLink"
			classes="u-margin-left-2x u-border-top-none u-border-bottom-none"
			[iconBefore]="IconName.edit"
			[link]="editLink"
			[type]="ButtonType.secondary"
		>
			{{ 'EDIT_COMPONENT__EDIT_FIELDS' | translate }}
		</ui-button>
	</div>
</nav>
