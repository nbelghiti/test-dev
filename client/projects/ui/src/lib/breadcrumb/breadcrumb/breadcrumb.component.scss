// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Breadcrumb
// =============================================================================
.c-breadcrumb {
	width: 100%;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-breadcrumb__list {
	@include extend-list-clear();

	display: flex;
}

.c-breadcrumb__content {
	display: flex;
	justify-content: space-between;
	margin: 0 auto;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-breadcrumb--border {
	border: $border-width-base solid $border-color-grey-86;
	border-right: none;
}
