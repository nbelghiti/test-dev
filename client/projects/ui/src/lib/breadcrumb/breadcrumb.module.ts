import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonModule } from '../button/button.module';
import { FilterModule } from '../filter/filter.module';
import { FlyoutModule } from '../flyout/flyout.module';
import { IconModule } from '../icon/icon.module';

import { BreadcrumbItemComponent } from './breadcrumb-item/breadcrumb-item.component';
import { BreadcrumbComponent } from './breadcrumb/breadcrumb.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		ButtonModule,
		RouterModule,
		FlyoutModule,
		FilterModule,
		ReactiveFormsModule,
		TranslateModule
	],
	declarations: [
		BreadcrumbItemComponent,
		BreadcrumbComponent,
	],
	exports: [
		BreadcrumbItemComponent,
		BreadcrumbComponent,
	],
})
export class BreadcrumbModule {}
