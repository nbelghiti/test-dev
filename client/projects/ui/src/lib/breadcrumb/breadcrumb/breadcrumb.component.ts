import { Component, Input } from '@angular/core';

import { ButtonType } from '../../button/button.types';
import { IconName } from '../../icon/icon.types';
import { IBreadcrumbItem } from '../breadcrumb.types';

@Component({
	selector: 'ui-breadcrumb',
	templateUrl: './breadcrumb.component.html',
	styleUrls: ['./breadcrumb.component.scss']
})

export class BreadcrumbComponent {
	@Input() public items: IBreadcrumbItem[] = [];
	@Input() public editLink: string[] = null;
	@Input() public first?: boolean = false;
	@Input() public border?: boolean = false;

	public IconName: typeof IconName = IconName;
	public ButtonType: typeof ButtonType = ButtonType;
}
