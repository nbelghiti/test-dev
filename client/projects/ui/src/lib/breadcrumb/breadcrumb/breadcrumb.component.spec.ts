import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonComponent } from '../../button/button/button.component';
import { FilterMenuItemComponent } from '../../filter/filter-menu-item/filter-menu-item.component';
import { FilterMenuComponent } from '../../filter/filter-menu/filter-menu.component';
import { IconComponent } from '../../icon/icon.component';
import { BreadcrumbItemComponent } from '../breadcrumb-item/breadcrumb-item.component';

import { BreadcrumbComponent } from './breadcrumb.component';

describe('BreadcrumbComponent', () => {
	let component: BreadcrumbComponent;
	let fixture: ComponentFixture<BreadcrumbComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				ReactiveFormsModule,
				TranslateModule.forRoot(),
			],
			declarations: [
				BreadcrumbComponent,
				IconComponent,
				BreadcrumbItemComponent,
				FilterMenuItemComponent,
				FilterMenuComponent,
				ButtonComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(BreadcrumbComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

