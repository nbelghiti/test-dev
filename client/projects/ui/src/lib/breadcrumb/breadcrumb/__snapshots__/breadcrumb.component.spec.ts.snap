// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbComponent Should render correctly 1`] = `
<ui-breadcrumb
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  border="false"
  editLink="null"
  first="false"
  items={[Function Array]}
>
  <nav
    class="c-breadcrumb"
  >
    <div
      class="c-breadcrumb__content"
    >
      <ul
        class="c-breadcrumb__list"
        ng-reflect-klass="c-breadcrumb__list"
        ng-reflect-ng-class="[object Object]"
      >
        
      </ul>
      
    </div>
  </nav>
</ui-breadcrumb>
`;
